frontend/src/component/project/Project/ExperimentalProjectFeatures/ExperimentalProjectTable/hooks/useRowActions.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import { FeatureArchiveDialog } from 'component/common/FeatureArchiveDialog/FeatureArchiveDialog';
+import { FeatureStaleDialog } from 'component/common/FeatureStaleDialog/FeatureStaleDialog';
+
+export const useRowActions = (onChange: () => void, projectId: string) => {
+    const [featureArchiveState, setFeatureArchiveState] = useState<
+        string | undefined
+    >();
+
+    const [featureStaleDialogState, setFeatureStaleDialogState] = useState<{
+        featureId?: string;
+        stale?: boolean;
+    }>({});
+
+    const rowActionsDialogs = (
+        <>
+            <FeatureStaleDialog
+                isStale={featureStaleDialogState.stale === true}
+                isOpen={Boolean(featureStaleDialogState.featureId)}
+                onClose={() => {
+                    setFeatureStaleDialogState({});
+                    onChange();
+                }}
+                featureId={featureStaleDialogState.featureId || ''}
```

Comment:
should we hide this dialog when no featureId or provide an empty value instead? Isn't empty values leading to invalid UI state?

Reply:
It is constructed in a way, where visibility of dialog component is passed as prop. Probably because of the pattern of how to deal with animations in React. It works how it worked before.